### ISSUE TITLE
show only specific day of week

### ISSUE THREAD

**Issue by Ich5003** [kizitonwose/CalendarView#233]
Hello,

is there any possibility to only show specific weekdays in the calender=
As an example, I need to create a calender, which only shows weekdays but no weekends.

**Comment by Ich5003** [comment]
Okay, I analyzed the code a bit, and it seems like this is currently not possible, as the number of days in a week is hardcoded, eg. here:

https://github.com/kizitonwose/CalendarView/blob/7fc0947ac626703f11e68d7adbc067946554fcd9/library/src/main/java/com/kizitonwose/calendarview/CalendarView.kt#L292

and here:

https://github.com/kizitonwose/CalendarView/blob/7fc0947ac626703f11e68d7adbc067946554fcd9/library/src/main/java/com/kizitonwose/calendarview/ui/CalendarAdapter.kt#L144

Treat this issue like a feature request.
I would love the see this possibility

**Comment by kizitonwose** [comment]
What is the use case for this feature exactly?

**Comment by Ich5003** [comment]
@kizitonwose 
Many people only work weekdays, so a view to only show the weekdays and hide the weekend would give them the possibility to make more use of the available screenspace for these 5 days.

**Comment by kizitonwose** [comment]
Guess I can close this as the feature is not supported and there's no planned support.

**Comment by Chrispassold** [comment]
I resolve this using the code below, the magic is `(widthSize / weekDaysCount) * 7`
This will prepare the view to create weekdays with only space for `weekDaysCount`

BUT, you must create an logic to remove the weekdays you don't want

```kotlin
internal class CustomCalendarView @JvmOverloads constructor(
    context: Context, attrs: AttributeSet, defStyleAttr: Int = 0
) : CalendarView(context, attrs, defStyleAttr) {

    // must be between 1 and 7
    var weekDaysCount: Int = 7
        private set

    init {

        context.withStyledAttributes(attrs, R.styleable.CustomCalendarView, defStyleAttr, 0) {
            weekDaysCount = getInt(R.styleable.CustomCalendarView_weekDaysCount, 7)
        }

        require(weekDaysCount in 1..7) {
            "Week days count must be equal or between 1 and 7"
        }

    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val widthMode = MeasureSpec.getMode(widthMeasureSpec)
        val widthSize = MeasureSpec.getSize(widthMeasureSpec)

        super.onMeasure(
            MeasureSpec.makeMeasureSpec((widthSize / weekDaysCount) * 7, widthMode),
            heightMeasureSpec
        )
    }
}
```

**Comment by Ich5003** [comment]
> 
> 
> I resolve this using the code below, the magic is `(widthSize / weekDaysCount) * 7`
> This will prepare the view to create weekdays with only space for `weekDaysCount`
> 
> BUT, you must create an logic to remove the weekdays you don't want
> 
> ```kotlin
> internal class CustomCalendarView @JvmOverloads constructor(
>     context: Context, attrs: AttributeSet, defStyleAttr: Int = 0
> ) : CalendarView(context, attrs, defStyleAttr) {
> 
>     // must be between 1 and 7
>     var weekDaysCount: Int = 7
>         private set
> 
>     init {
> 
>         context.withStyledAttributes(attrs, R.styleable.CustomCalendarView, defStyleAttr, 0) {
>             weekDaysCount = getInt(R.styleable.CustomCalendarView_weekDaysCount, 7)
>         }
> 
>         require(weekDaysCount in 1..7) {
>             "Week days count must be equal or between 1 and 7"
>         }
> 
>     }
> 
>     override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
>         val widthMode = MeasureSpec.getMode(widthMeasureSpec)
>         val widthSize = MeasureSpec.getSize(widthMeasureSpec)
> 
>         super.onMeasure(
>             MeasureSpec.makeMeasureSpec((widthSize / weekDaysCount) * 7, widthMode),
>             heightMeasureSpec
>         )
>     }
> }
> ```

Thanks, I also already solved it and even created a PR (see #237 ) but it was rejected because "this introduces a whole lot of complexity that may not be desirable(maintenance-wise) in the future."

If you need a working version where you can already set the weekdays you desire you could try my fork of this repository https://github.com/Ich5003/CalendarView-WeekdayFilter

**Comment by Chrispassold** [comment]
> > I resolve this using the code below, the magic is `(widthSize / weekDaysCount) * 7`
> > This will prepare the view to create weekdays with only space for `weekDaysCount`
> > BUT, you must create an logic to remove the weekdays you don't want
> > ```kotlin
> > internal class CustomCalendarView @JvmOverloads constructor(
> >     context: Context, attrs: AttributeSet, defStyleAttr: Int = 0
> > ) : CalendarView(context, attrs, defStyleAttr) {
> > 
> >     // must be between 1 and 7
> >     var weekDaysCount: Int = 7
> >         private set
> > 
> >     init {
> > 
> >         context.withStyledAttributes(attrs, R.styleable.CustomCalendarView, defStyleAttr, 0) {
> >             weekDaysCount = getInt(R.styleable.CustomCalendarView_weekDaysCount, 7)
> >         }
> > 
> >         require(weekDaysCount in 1..7) {
> >             "Week days count must be equal or between 1 and 7"
> >         }
> > 
> >     }
> > 
> >     override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
> >         val widthMode = MeasureSpec.getMode(widthMeasureSpec)
> >         val widthSize = MeasureSpec.getSize(widthMeasureSpec)
> > 
> >         super.onMeasure(
> >             MeasureSpec.makeMeasureSpec((widthSize / weekDaysCount) * 7, widthMode),
> >             heightMeasureSpec
> >         )
> >     }
> > }
> > ```
> 
> Thanks, I also already solved it and even created a PR (see #237 ) but it was rejected because "this introduces a whole lot of complexity that may not be desirable(maintenance-wise) in the future."
> 
> If you need a working version where you can already set the weekdays you desire you could try my fork of this repository https://github.com/Ich5003/CalendarView-WeekdayFilter

I will take a look :)
My suggestion is not 100%, im having problems with empty weekdays :P

![image](https://user-images.githubusercontent.com/6041579/116291070-90a47100-a76a-11eb-8ec6-474eb355a3b8.png)
